### PR TITLE
Teach Docker to Auto-GPT

### PR DESCRIPTION
**Pull request by finaldzn**
Work in progress, but im Trying to give auto gpt a way to build, manage, and run command inside docker containers.

The goal are : 
- No need tp set up any dev environnment for Auto-Gpt to work
- A way for Auto-GPT to run any commands in a controlled environnement
- A way for Auto-GPT to develop complete apps using the virtual docker environment.

Im not sure about : 
- The prompts given to explain the commands
- The goal i gave auto gpt

Here is the reference goal with wich i've tried : 

Any suggestions is welcomed

```
ai_goals:
- set up a dev environnement for a react app using docker
- 'create a new bare react app '
- add a component to the react app that shows the current time
- run the app, access it and print in a file the html
ai_name: Developer-GPT
ai_role: Set up working coding a environment and then follow user instructions to
  create an app
```

### CODE REVIEW

File: scripts/docker_test.py
Comment (st4s1k) [inline scripts/docker_test.py:1]: shouldn't a test assert something?
at least print out the result of an equals operation

File: scripts/commands.py [hunk 1]
```diff
@@ -19,15 +20,15 @@
 def get_command(response):
     try:
         response_json = fix_and_parse_json(response)
-
```
Comment (nponeccop) [inline scripts/commands.py:22]: Submit the whitespace fixes as a separate PR, to keep the diff here small.

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by nponeccop** [comment]
@finaldzn Now it's outdated. Resolve the conflicts and make it more focused

**Comment by AndresCdo** [comment]
@finaldzn Resolve the conflicts.

File: scripts/commands.py [hunk 2]
```diff
@@ -117,6 +128,7 @@ def google_search(query, num_results=8):
 
     return json.dumps(search_results, ensure_ascii=False, indent=4)
 
+
```
Comment (AndresCdo) [inline scripts/commands.py:131]: Delete whitespace.

File: scripts/commands.py [hunk 3]
```diff
@@ -152,6 +165,7 @@ def google_official_search(query, num_results=8):
     # Return the list of search result URLs
     return search_results_links
 
+
```
Comment (AndresCdo) [inline scripts/commands.py:168]: Delete whitespace.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
Closing this PR because no activity in the past 2 months. Interesting idea though; I have labeled it with `fridge` so we can find it back and use it as inspiration later, when the dust of re-arch settles. Or if you want, feel free to rebase your changes on current `master` and resubmit :)

**Pull request closed without merging**